hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/JoinCheckOperation.java

```diff
@@ -61,12 +66,13 @@ public void run() {
             ILogger logger = getLogger();
             try {
                 if (service.getClusterJoinManager().validateJoinMessage(request)) {
-                    try {
-                        // also validate other node's version is able to join our cluster
-                        node.getNodeExtension().validateJoinRequest(request);
+                    // also validate other cluster's major.minor version is same as this cluster
```

Comment:
Could we add a comment, why they have to be the same, e.g. it will lead to sneaky-downgrade or sneaky-upgrade?
We may simply forget it in the future...